packages/py-moose-lib/moose_lib/query_param.py

```diff
@@ -175,3 +175,24 @@ def parse(param: str, t: scalar_types) -> Any:
             [v] = values
             constructor_args[field_name] = parse(v, field_type)
     return cls(**constructor_args)
+
+def normalize_query_params(query_params: dict[str, list[str]]) -> dict[str, Any]:
+    """
+    Normalize query parameters by unwrapping single-item lists to scalar values.
+
+    When query parameters are parsed with parse_qs(), all values are returned as lists,
+    even when there's only one value. This function unwraps single-item lists while
+    preserving multi-value lists.
+
+    Examples:
+    - {'color': ['red']} becomes {'color': 'red'}
```

Comment:
What if `color` really is a list field? then the validation will fail.

what's wrong with using `map_params_to_class`?

Reply:
Hm, how do I use it with ConsumptionApi class? It looks specific to how dmv1 is structured.

```
def map_params_to_class(
        params: dict[str, list[str]],
        field_def_list: list[QueryField],
        cls: type,
) -> Any:
```

Reply:
The `ConsumptionApi` class has `columns: Columns[T]` inherited from `TypedMooseResource`. That function needs `list[QueryField]`.

But that feels weird, a consumption API does not have columns.

Reply:
Gotcha. I did some refactoring, lmk if that's what you had in mind.